MCDataProducts/inc/CaloClusterMC.hh

```diff
@@ -1,38 +1,34 @@
 #ifndef MCDataProducts_CaloClusterMC_hh
 #define MCDataProducts_CaloClusterMC_hh
-//
-// Summarize the truth content of a CaloCluster
-//
-#include "canvas/Persistency/Common/Assns.h"
-#include "canvas/Persistency/Common/Ptr.h"
-#include "MCDataProducts/inc/SimParticle.hh"
-#include "MCDataProducts/inc/MCRelationship.hh"
-#include "RecoDataProducts/inc/CaloCluster.hh"
+
+#include "MCDataProducts/inc/CaloEDepMC.hh"
 #include <vector>
-#include <utility>
-namespace mu2e {
-  struct CaloMCEDep {
-    art::Ptr<SimParticle> const& sim() const { return _simp; }
-    float energyDeposit() const { return _edep; }
-    float time() const { return _time; }
-    art::Ptr<SimParticle> _simp; // sim particle
-    MCRelationship _rel; // relationship of this particle to its primary
-    float _edep; // energy deposited by this particle in this cluster
-    float _time; // average time of the energy deposition by this particle; includes all offsets!
-    // should include an energy-weighted position FIXME!
-  };
-    
-  struct CaloClusterMC {
-    std::vector<CaloMCEDep> const& energyDeposits() const { return _edeps; }
-    float totalEnergyDeposit() const { return _edep; }
-    float averageTime() const { return _time; }
-    
-    std::vector<CaloMCEDep> _edeps; // energy contributions from individual SimParticles, sorted by contribution
-    float _edep; // total edep
-    float _time; // energy-weighted time average, including all offsets!
-    // should include an energy-weighted position FIXME!
-  };
-  typedef std::vector<mu2e::CaloClusterMC> CaloClusterMCCollection;
-  typedef art::Assns<CaloCluster,CaloClusterMC> CaloClusterMCAssns;
-}
+
+namespace mu2e
+{   
+   
+   class CaloClusterMC
+   {
+       public:                    
+          CaloClusterMC()                                     : edeps_()      {};
+          CaloClusterMC(const std::vector<CaloEDepMC>& edeps) : edeps_(edeps) {};
+                
+                std::vector<CaloEDepMC>&   energyDeposits  ()                 {return edeps_; }
+                CaloEDepMC&                energyDeposit   (unsigned i)       {return edeps_.at(i); } 
+          const std::vector<CaloEDepMC>&   energyDeposits  ()           const {return edeps_; }
```

Comment:
Just to be sure, you only want to save the steps if their energy deposited is above a threshold, or you only want to return the steps above a certain energy deposited but save them all anyway.

Reply:
TBH, I don't recall any use case where the users had to inspect the energy deposits of the particles with E<50 MeV. BUT maybe now for the CAPHRI studies they need them... Maybe you can configure the truth matching module with a data member that set this threshold?

Reply:
I'm getting very confused about what you would like to have. Currently, we save the energy deposits for each incoming SimParticle, and I think we need to keep doing this. However, some of the energy deposits have very small value. Do you propose to avoid saving them, or save them but provide a method that will list the energy deposits only above some threshold?

Reply:
Are you thinking of the 50 MeV threshold used in compression to suppress unassociated energy deposits?  Compression saves all energy deposits associated with a reconstructed track or primary particle.

Reply:
I apologize for the confusion. Let me try to clarify; in the previous version of the code, there was this class: 
https://github.com/Mu2e/Offline/blob/master/CaloMC/inc/ClusterContentMC.hh
during the summer, I modified that class with one of the students from my group:
/mu2e/app/users/gianipez/dev_Offline/CaloMC/inc/ClusterContentMC.hh
I thought the energyDeposits would correspond to the grouped energy deposited by single particles. Is this assumption correct? if so, I would put a threshold to that energy to avoid having a ton of neutrons in that list.

Reply:
The new code contain a vector of MC objects for each reconstructed cluster. These objects contain the amount of energy deposited and the PdgID of the associated SimParticle, ordered by energy deposited. If you are interested to the main contributors, you only need to look at the first (maybe second) entries in that vector. 
As an aside, I have also added the possibility to flag SimParticles leaving very small amount of energy (threshold as parameter) in the crystal, so that Andy's compression code can remove them.